torch/csrc/jit/tracer.cpp

```diff
@@ -106,15 +106,20 @@ Node* recordTrace(std::string op, // TODO: make this a Symbol
   // NB: Order matters. This must append after inputs but before outputs.
   graph->appendNode(n);
 
-  int i = 0;
-  for (Variable output : outputs) {
-    Node* sel = graph->appendNode(graph->createSelect(n, i));
-    // TODO: Track inplace operations (needed for JIT).
+  auto assignOutput = [&state](const Variable & output, Node * value) {
     if (output.defined()) {
-      sel->inferTypeFrom(output.data());
-      setValueTrace(state, output, sel);
+      value->inferTypeFrom(output.data());
+      setValueTrace(state, output, value);
+    }
+  };
+  if(outputs.size() == 1) {
+    assignOutput(outputs[0],n);
+  } else {
+    int i = 0;
+    for (Variable output : outputs) {
+      assignOutput(output, graph->appendNode(graph->createSelect(n, i)));
+      i++;
```

Comment:
Can you leave this for a separate commit? I think there were some places in Ed's latest commits (now in master), that assumed we never have single-return ops

Reply:
also nit: use a for loop if you need the index

Reply:
We can't leave it out, it would require extensive changes in the fusion passes and the fusion compiler that we would just need to change back. I fixed _most_ of the onnx issues, I believe. Moving the ONNX passes out of the repo prevents fixing all of them.